addons/im_livechat/controllers/main.py

```diff
@@ -65,6 +65,8 @@ def livechat_init(self, channel_id):
         # find the first matching rule for the given country and url
         matching_rule = request.env['im_livechat.channel.rule'].sudo().match_rule(channel_id, url, country_id)
         if matching_rule and (not matching_rule.chatbot_script_id or matching_rule.chatbot_script_id.script_step_ids):
+            frontend_lang = request.env.user._is_public() and request.httprequest.cookies.get('frontend_lang') or request.env.user.lang or 'en_US'
```

Comment:
What is the reason to call `_is_public` instead of this?
```suggestion
            frontend_lang = request.httprequest.cookies.get('frontend_lang', request.env.user.lang or 'en_US')
```

Reply:
@tsm-odoo  Thanks for the check!
I wasn't sure about the right behavior, so I opted for:
- Not logged: use website language (if set, otherwise default public user language)
- Logged: use user language

In your case, the website language will be the one that prevails regardless if logged or not... I think it's better indeed, updating the PR, thanks!